### PR TITLE
Add systemd support

### PR DESCRIPTION
**Pull request by davidmhewitt**
Adds the necessary unit files to be able to launch Pantheon via `gnome-session`'s systemd support.

Merging this will have no effect on the current way of launching the session, just prepares us for removing the `--builtin` flag on the session in the future.

This can be tested in conjunction with https://github.com/elementary/gala/pull/1256 and https://github.com/elementary/gala/pull/1255 . If you remove the `--builtin` flag from `/usr/share/xsessions/pantheon.desktop`, `gnome-session` will launch Pantheon using the systemd units instead of the XDG autostarts.

### CODE REVIEW

**Comment by davidmhewitt** [comment]
@bobby285271 can you have a look at this?

We seem to have nix packaging in-tree here and I've broken it :sweat_smile: 

This PR adds some systemd user units that along with some changes to gala (and probably some other components) will resolve https://github.com/elementary/session-settings/issues/17

I think pantheon elementary OS 6.0 will still continue to use the `--builtin` flag on `gnome-session` (i.e. it won't use these systemd units and use XDG autostarts and the builtin session mangement in `gnome-session` instead), but there's no harm in installing these in preparation for the future. Or you could just disable them with the meson option for now.

**Comment by bobby285271** [comment]
https://github.com/bobby285271/session-settings/actions/runs/1250907248
https://github.com/bobby285271/session-settings/commit/9f1aa216205143fd6f9a9dd9bf5ab0d7a872e0ba

This should be the minimal change to make CI pass, though I was not able to `nix-build` this locally now (due to https://github.com/NixOS/nixpkgs/pull/108934 I think). I think `systemd` should probably be put in `buildInputs`, ping @davidak for some review :-)

**Comment by davidmhewitt** [comment]
@bobby285271 Thanks, I've made those changes. I've never done anything with nix packaging before, so if you and @davidak are happy, I'm happy. There's a relevant issue about this change in NixOS here: https://github.com/NixOS/nixpkgs/issues/101817

**Comment by davidak** [comment]
I see no problem, but i'm also no expert with session management.

**Comment by davidmhewitt** [comment]
That's OK, I'll handle testing the session management stuff, just wanted to check the nix packaging files still look good!

File: systemd/gnome-session-x11@pantheon.target
Comment (JoseExposito) [inline systemd/gnome-session-x11@pantheon.target:9]: Should we remove this comments before merging?

**Review by JoseExposito — DISMISSED** [review]
Not an expert, but from my limited experience on the topic, it looks good. I'd appreciate if someone else could look at it as well.

File: systemd/gnome-session-x11@pantheon.target
Comment (davidmhewitt) [reply systemd/gnome-session-x11@pantheon.target:9]: These unit files are just direct copies of the ones from `gnome-session`, just tweaked where necessary to make it run pantheon instead and these lines come from there. It seems nicer to keep them as similar as possible to the `gnome-session` ones so we can see the diff if necessary.

**Comment by davidak** [comment]
I tested this on NixOS without issues. https://github.com/NixOS/nixpkgs/pull/140429#issuecomment-984217202

We will merge it into our unstable release when this PR is merged.

When do you plan to merge this PR?

**Comment by bobby285271** [comment]
Just for information, we later found some issues with `--builtin` removed like hotkeys for screenshot, keyboard keys for volume up, down and mute, night light no longer works. But yeah this PR does not actually do the switch so should be fine.

**Comment by davidmhewitt** [comment]
> Just for information, we later found some issues with `--builtin` removed like hotkeys for screenshot, keyboard keys for volume up, down and mute, night light no longer works. But yeah this PR does not actually do the switch so should be fine.

That sounds like the various GSD components aren't starting as dependencies, did you find the source of the issue?

**Comment by bobby285271** [comment]
> That sounds like the various GSD components aren't starting as dependencies, did you find the source of the issue?

Just updated the downstream PR to include the workaround: https://github.com/NixOS/nixpkgs/pull/140429#discussion_r780134906 Sorry for the noise :crying_cat_face:

**Review by tintou — APPROVED** [review]

**Pull request merged**
